tools/ClangTidyAll.sh

```diff
@@ -8,8 +8,53 @@ ALL_SOURCE_FILES=`find \
                   @CMAKE_SOURCE_DIR@/tests \
                   -name "*.cpp"`
 
-for file in $ALL_SOURCE_FILES
+if [ ${#ALL_SOURCE_FILES[@]} == 0 ]; then
+   echo "No C++ files found in repository"
+   exit 0
+fi
+
+BUILD_DIR=@CMAKE_BINARY_DIR@
+
+if [ $# != 0 ] && [ $# != 1 ]; then
+    echo "Wrong number of arguments passed to ClangTidyAll.sh, $#"
+    echo "Expecting none or just the number of threads to run on."
+    exit 1
+fi
+
+THREADING_FLAG="-j 1"
+if [ $# == 1 ]; then
+    THREADING_FLAG="-j $1"
+fi
+
+# Try to find run-clang-tidy-LLVM_VERSION in the path.
+RUN_CLANG_TIDY_BIN=
+for dir in $(echo ${PATH} | tr ':' '\n');
 do
-    printf "\n\nChecking file: $file\n"
-    make clang-tidy FILE=$file
+    if [ -d $dir ]; then
+        RUN_CLANG_TIDY_BIN=`find $dir -name run-clang-tidy* | head -n 1`
```

Comment:
Good! The star is important.  Sometimes there is run-clang-tidy-LLVM_VERSION.py and not run-clang-tidy.py (but maybe this is a problem only for old LLVM).